refactor(esco-content-menu): use the portletRegistry API

jgribonvald opened the pull request:
use of the portletRegistry API instead of the marketPlace API

ChristianMurphy commented on @uportal/esco-content-menu/src/services/flattenPortlets.js:
```diff
@@ -0,0 +1,47 @@
+export default function flattenPortlets(elem) {
```
Maybe this could leverage https://github.com/uPortal-contrib/uPortal-web-components/tree/master/%40uportal/portlet-registry-to-array ?

jgribonvald replied:
done, but with some local change and so moved locally

ChristianMurphy commented:
@jgribonvald is this still in progress?

jgribonvald commented:
@ChristianMurphy: yes, sorry I were out of office and i'm back only now and have several things to manage before ...

drewwills commented:
Folks -- I suspect the `esco-content-menu` could get a lot of adoption at the Winter Summit if we could get this change merged before then.

drewwills commented:
Thanks @jgribonvald !

I'm hoping to try these changes in the next couple days with this PR:  https://github.com/Jasig/uPortal-start/pull/220

jgribonvald commented:
I fixed all bugs on this parts, I didn't see anymore things... please provide a feedback !

One important thing: I used a custom portletRegistryToArray, where I added a part to merge all categories into portlets informations with a custom uniqueBy function due to that.

jgribonvald commented on @uportal/esco-content-menu/.env:
```diff
@@ -4,7 +4,7 @@ VUE_APP_PORTAL_BASE_URL=
 VUE_APP_FAVORITE_ADD_URI=/api/layout?action=addFavorite&channelId=
 VUE_APP_FAVORITE_RM_URI=/api/layout?action=removeFavorite&channelId=
-VUE_APP_BROWSABLE_PORTLETS_URI=/api/marketplace/entries.json
+VUE_APP_BROWSABLE_PORTLETS_URI=/api/v4-3/dlm/portletRegistry.json?category=All%20categories
```
I fixed this part by removing the extra part !

jgribonvald commented:
Is it ok for everyone to merge it ?

The only remainin thing to optimize would be about favorites, in my case (inside hamburger-menu/content-menu), I need to get the favorites order whereas it's not the case inside content-grid, so we could have a request less (to layout.json) in this case.
Maybe an improvment inside portletRegistry would be to provide a portlet order attribute inside portlet definition ? going into uportl-dev list to talk about that ? or creating an issue ?

drewwills commented:
Maybe an improvment inside portletRegistry would be to provide a portlet order attribute inside portlet definition?

Metadata that helps with sorting would be a sensible addition.

But I'm not sure if we would need to change the fields in the JSON -- we can already do it with publishing attributes.
